hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -996,77 +996,289 @@
+    //          If this is set to False, the pad can be referenced by direct connections in
+    //          the pinmux signal mapping further below.
+    //          If an unmuxed pad is NOT referenced in the pinmux signal mapping,
+    //          the pad needs to be wired up manually in the template.
+    //
+    // Optionally, each pad can also have a 'desc' field for further description.
+    pads: [
+      // TODO: this needs to be updated to the latest ASIC pinout.
+      // Dedicated
+      { name: 'POR_N'           , type: 'InputStd', bank: 'VCC' , muxed: 'False', desc: 'System reset'},
+      { name: 'SPI_HOST_D0'     , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI host data'},
+      { name: 'SPI_HOST_D1'     , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI host data'},
+      { name: 'SPI_HOST_D2'     , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI host data'},
+      { name: 'SPI_HOST_D3'     , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI host data'},
+      { name: 'SPI_HOST_CLK'    , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI host clock'},
+      { name: 'SPI_HOST_CS_L'   , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI host chip select'},
+      { name: 'SPI_DEV_D0'      , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI device data'},
+      { name: 'SPI_DEV_D1'      , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI device data'},
+      { name: 'SPI_DEV_D2'      , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI device data'},
+      { name: 'SPI_DEV_D3'      , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI device data'},
+      { name: 'SPI_DEV_CLK'     , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI device clock'},
+      { name: 'SPI_DEV_CS_L'    , type: 'InputStd', bank: 'VIOA', muxed: 'False', desc: 'SPI device chip select'},
+      { name: 'USB_P'           , type: 'BidirTol', bank: 'VCC' , muxed: 'False', desc: 'USB P signal'},
+      { name: 'USB_N'           , type: 'BidirTol', bank: 'VCC' , muxed: 'False', desc: 'USB N signal'},
+      { name: 'CC1'             , type: 'InputStd', bank: 'AVCC', muxed: 'False', desc: 'ADC input 1'},
+      { name: 'CC2'             , type: 'InputStd', bank: 'AVCC', muxed: 'False', desc: 'ADC input 2'},
+      { name: 'FLASH_TEST_MODE0', type: 'InputStd', bank: 'VCC' , muxed: 'False', desc: 'Flash test mode signal'},
+      { name: 'FLASH_TEST_MODE1', type: 'InputStd', bank: 'VCC' , muxed: 'False', desc: 'Flash test mode signal'},
+      { name: 'FLASH_TEST_MODE2', type: 'InputStd', bank: 'VCC' , muxed: 'False', desc: 'Flash test mode signal'},
+      { name: 'FLASH_TEST_MODE3', type: 'InputStd', bank: 'VCC' , muxed: 'False', desc: 'Flash test mode signal'},
+      { name: 'FLASH_TEST_VOLT' , type: 'InputStd', bank: 'VCC' , muxed: 'False', desc: 'Flash test voltage input'},
```

Comment:
i've fired off an email, but i think this one might need to be handled slightly differently. stay tuned..

Reply:
Ok, no worries. 
There are some more fixed wrt to the pinout itself that needs to go in after this PR.